### PR TITLE
NETBEANS-6307: check global artifact cache, ignore per-project cached  structure if not consistent.

### PR DESCRIPTION
**Pull request by sdedic**
For detailed explanation, see [bug description](https://issues.apache.org/jira/browse/NETBEANS-6305). The fix is actually a workaround - the cached project structure/information is validated against global artifact cache. The cached info is rejected if the global artifact cache does not contain items that would have been present/updated if the current IDE instance had loaded the project fully and created the caches (both global and project).

A test is attached that demonstrates the issue, if [the validation call](https://github.com/apache/netbeans/pull/3375/files#diff-e0b2de3f6a585632d92c8419e75f246740dcb1f90a52b33e4a58147423b69bdbL45) is removed.

### CODE REVIEW

**Comment by sdedic** [comment]
72hours elapsed; reviewers, please comment / suggest improvements. Otherwise I'll merge in the change on 21th Dec.

File: extide/gradle/src/org/netbeans/modules/gradle/loaders/GradleArtifactStore.java [hunk 1]
```diff
@@ -159,7 +171,54 @@ void verify() {
             store();
         }
     }
-
+    
+    /**
+     * Checks that all dependencies the project thinks should be in the global cache
+     * are actually in the global cache. If the global artifact cache does not contain
+     * an entry from a resolved depency in the project cache then many random failures can
```
Comment (JaroslavTulach) [inline extide/gradle/src/org/netbeans/modules/gradle/loaders/GradleArtifactStore.java:178]: typo: dependency

File: extide/gradle/test/unit/src/org/netbeans/modules/gradle/loaders/DiskCacheProjectLoaderTest.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                        addTest(DiskCacheProjectLoaderTest.class).
+                        addTest("testLoadCachedProject").
+                        reuseUserDir(false).
+                        gui(false).
+                        enableModules(".*gradle.*").
+                        reuseUserDir(false);
+        
+        Test run2 = NbModuleSuite.create(conf);
+        ts.addTest(run1);
+        ts.addTest(run2);
```
Comment (JaroslavTulach) [inline extide/gradle/test/unit/src/org/netbeans/modules/gradle/loaders/DiskCacheProjectLoaderTest.java:81]: A test with a restart of the IDE! Amazing.

**Review by JaroslavTulach — APPROVED** [review]
Great work creating the test. That must mean something has been fixed!

**Review by entlicher — APPROVED** [review]

**Comment by sdedic** [comment]
Transient errors went off, merging.

**Pull request merged**